hangupsbot/handlers.py

```diff
@@ -47,23 +47,84 @@ def __init__(self, bot, bot_command='/bot'):
                              {},
                              forgiving=True )
 
-    def register_handler(self, function, type="message", priority=50):
-        """registers extra event handlers"""
+    def register_handler(self, function, type="message", priority=50, extra_metadata={}):
```

Comment:
Mutable default argument for `extra_metadata` will be shared across function calls.

Currently this _shouldn't_ bite us -- we only store an original reference to it during a late registration, but a non-default should always be supplied for that -- but probably better to be explicit, e.g. `extra_metadata=None` argument coupled with `extra_metadata = extra_metadata or {}` or similar.